dom-fetch-client/src/main/scala/org/http4s/dom/FetchReferrer.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.dom
+
+import org.http4s.Uri
+import org.http4s.util.Writer
+import org.http4s.util.Renderable
+
+sealed trait FetchReferrer extends Renderable
+object FetchReferrer {
```

Comment:
This was definitely the right thing to do :)
```suggestion
sealed abstract class FetchReferrer extends Renderable
object FetchReferrer {
```

Reply:
No problem with making this change, but I'm curious why an `abstract class` is preferable to a `trait` in this case?

Reply:
Thanks. Eh, just my own superstitious voodoo 😆 the binary encoding and runtime behavior of `trait`s is/was slightly more complex than an `abstract class`. So I prefer to be conservative for `sealed` hierarchies since we have full control and only reach for `trait` if necessary.

Reply:
@armanbilge I'm of the same opinion as you, though I think on Scala >= 2.12.x if it is sealed then the differences are very minor between an `abstract class` and a `trait`. The story was different on 2.11.x when Scala wasn't using the JVM based default methods in interfaces.